src/indexer/json_term_writer.rs

```diff
@@ -260,12 +260,8 @@ pub struct JsonTermWriter<'a> {
 }
 
 impl<'a> JsonTermWriter<'a> {
-    pub fn from_field_and_json_path(
-        field: Field,
-        json_path: &str,
-        term_buffer: &'a mut Term,
-    ) -> Self {
-        term_buffer.set_field(Type::Json, field);
+    pub fn from_json_path(json_path: &str, field: Field, term_buffer: &'a mut Term) -> Self {
```

Comment:
why change the order?

field is coming first logically and in the representation

Reply:
is it? `from_json_path` and json_path was the argument in the middle. field is only used for the term_buffer

Reply:
Yes.

Representation is:
Field / Type / json_path / value

Logic is 
Field > json_path > value